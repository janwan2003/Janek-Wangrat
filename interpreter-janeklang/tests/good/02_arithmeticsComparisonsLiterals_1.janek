int main() {
    10;
    "string";
    true;
    int a = 10;
    int b = 5;
    int c = a + b;
    print(c);
    c = a - b;
    print(c);
    c = a * b;
    print(c);
    c = a / b;
    print(c);
    c = a % b;
    print(c);
    if (a > b) {
        print("a is greater than b");
    }
    if (a < b) {
        print("a is less than b");
    } else {
        print("a is not less than b");
    }
    if (a == 10) {
        print("a is 10");
    }
    if (b != 10) {
        print("b is not 10");
    }
    if (a >= b) {
        print("a is greater than or equal to b");
    }
    if (b <= a) {
        print("b is less than or equal to a");
    }
    if ((a + b) > (a - b)) {
        print("a + b is greater than a - b");
    }
    if ((a > b) && (b > 0)) {
        print("a is greater than b and b is positive");
    }
    if ((a < b) || (a > b)) {
        print("a is either less than or greater than b");
    }
    bool d = true;
    bool e = false;
    if (d) {
        print("d is true");
    }
    if (!e) {
        print("e is false");
    }
    if (d && e) {
        print("d and e are true");
    } else {
        print("d and e are not both true");
    }
    if (d || e) {
        print("d or e is true");
    }
    return 0;
}
