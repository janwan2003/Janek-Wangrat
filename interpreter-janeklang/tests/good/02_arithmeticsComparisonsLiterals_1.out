15
5
50
2
0
a is greater than b
a is not less than b
a is 10
b is not 10
a is greater than or equal to b
b is less than or equal to a
a + b is greater than a - b
a is greater than b and b is positive
a is either less than or greater than b
d is true
e is false
d and e are not both true
d or e is true
